customer-endpoint/src/main/kotlin/org/ostelco/prime/customer/endpoint/resources/ProductsResource.kt

```diff
@@ -54,4 +54,31 @@ class ProductsResource(private val dao: SubscriberDAO) {
                         saveCard = saveCard ?: false)
                         .responseBuilder(Response.Status.CREATED)
             }.build()
+
+    @POST
+    @Path("{sku}/renew")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun renewPaymentSubscription(@Auth token: AccessTokenPrincipal?,
+                                 @NotNull
+                                 @PathParam("sku")
+                                 sku: String,
+                                 @QueryParam("sourceId")
+                                 sourceId: String?,
+                                 @QueryParam("saveCard")
+                                 saveCard: Boolean?): Response =
+            if (token == null) {
```

Comment:
Can authorisation check be delegated to a filter so that it does not show up in the business logic part of the resource implementation?  Technically it obviously can, but the question is obviously if it makes sense in this particular situation.  If  the "token == null"  must be there, add a comment describing why this couldn't be delegated to a filter, of if it can, at the very least add a "TODO" indicating that it should.

Reply:
Agree with this. I think the same pattern is used in all client facing APIs, and an update should probably fix it for all of them. Will leave a TODO and a Notion tech dept task for fixing this.